embedded/sql/engine.go

```diff
@@ -937,15 +937,16 @@ func EncodeValue(val interface{}, colType SQLValueType, maxLen int) ([]byte, err
 		}
 	case IntegerType:
 		{
-			intVal, ok := val.(uint64)
+			intVal, ok := val.(int64)
 			if !ok {
 				return nil, ErrInvalidValue
 			}
 
+			// map to unsigned integer space
 			// len(v) + v
 			var encv [EncLenLen + 8]byte
 			binary.BigEndian.PutUint32(encv[:], uint32(8))
-			binary.BigEndian.PutUint64(encv[EncLenLen:], intVal)
+			binary.BigEndian.PutUint64(encv[EncLenLen:], uint64(intVal+math.MaxInt64+1))
```

Comment:
Same here, mapping from signed to unsigned space can be done by just a simple type conversion, no need to shift values:

```suggestion
			binary.BigEndian.PutUint64(encv[EncLenLen:], uint64(intVal))
```

Reply:
replied above